src/data_provider/src/packages/packageLinuxParserApk.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Wazuh SYSINFO
+ * Copyright (C) 2015, Wazuh Inc.
+ * December 19, 2022.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "json.hpp"
+#include <fstream>
+
+void getApkInfo(const std::string& fileName, std::function<void(nlohmann::json&)> callback)
+{
+    std::ifstream apkDb(fileName);
+    const size_t valuePos {2};
+    std::string line;
+    std::vector<std::string> data;
+
+    if (apkDb.is_open())
+    {
+        while (getline(apkDb, line))
+        {
+            if (!line.empty())
+            {
+                switch (line.front())
+                {
+                    case 'P':
+                        data.push_back(line.substr(valuePos));
+                        break;
+
+                    case 'V':
+                        data.push_back(line.substr(valuePos));
+                        break;
+
+                    case 'A':
+                        data.push_back(line.substr(valuePos));
+                        break;
+
+                    case 'I':
+                        data.push_back(line.substr(valuePos));
+                        break;
+
+                    case 'T':
+                        data.push_back(line.substr(valuePos));
+                        break;
+                }
+            }
+            else
+            {
+                nlohmann::json packageInfo;
+                packageInfo["name"] = data.at(0);
```

Comment:
This implementation requires UT to be validated.
For example, if there is any expected letter missing for a package, the program will thrown an exception like this one
```
terminate called after throwing an instance of 'std::out_of_range'
    what():  vector::_M_range_check: __n (which is 4) >= this->size() (which is 4)
```

Other possibility is that the file could be corrupted or being written in the exact moment the agent is reading it.
For example, if any of the letter doesn't have a `:` we'll have another exception

```
terminate called after throwing an instance of 'std::out_of_range'
  what():  basic_string::substr: __pos (which is 2) > this->size() (which is 1)
```

Reply:
Added UTs